### ISSUE TITLE
Shouldn't function values be run no matter if there are params or not?

### ISSUE THREAD

**Issue by joakimbeng** [staltz/switch-path#6]
I think the api is inconsistent when using functions as route values. They behave one way for routes without params and another way when there are params.

I'm using `switch-path` together with `cycle-history` and would like the route values to lazily evaluate to Cycle components (i.e. wrap them in functions). Which at the moment looks like this:

``` javascript
const page$ = History.map(({pathname}) => {
  const {value} = switchPath(pathname, {
    '/items': () => ItemsComponent({DOM}),
    '/item/:id': id => () => ItemComponent({DOM, props$: Rx.Observable.just({id})}),
    '*': () => Home({DOM})
  });
  return value().DOM;
})
.mergeAll();
```

You see, for the param route the value needs to be a function which returns a function, this is to have a consistent return value in the `map` above.

What I instead would like to do is:

``` javascript
const page$ = History.map(({pathname}) => {
  const {value} = switchPath(pathname, {
    '/items': () => ItemsComponent({DOM}),
    // no more need to have double functions below:
    '/item/:id': id => ItemComponent({DOM, props$: Rx.Observable.just({id})}),
    '*': () => Home({DOM})
  });
  return value.DOM;  // the value function is already run no matter if it has params or not
})
.mergeAll();
```

What do you think?

**Comment by staltz** [comment]
Hi @joakimbeng. Thank you for suggesting. This is an interesting use case. But we need to remember that switchPath is not made for Cycle.js only, it is a generic tool. Essentially it is just a function that lets people do pattern matching of URLs more easily. I think the double function is much more clear and has less magic to it. Plus, it's a very small cost to pay (just type 4 more characters: `() =>`).

Try to look at it from the perspective of someone who is reading your code for the first time. The first option is clear what it does.

**Comment by joakimbeng** [comment]
I understand your decision. Though I disagree on the first option being more clear. Because for me it wasn't clear that this module only runs functions for matches with parameters, I had to look at the source code to catch that. Also when not using arrow functions (which still is the most common scenario when talking about generic tools IMO) it's not just 4 more characters to type ;)

I decided to build a module heavily inspired by this module but fits my needs better: [`handle-path`](https://www.npmjs.com/package/handle-path). The differences between switch-path and handle-path are listed in the readme.

And by the way thank you for your great work with Cycle.js! :)
